include/seqan3/range/view/async_buffer.hpp

```diff
@@ -0,0 +1,538 @@
+        {
+            return !(lhs == std::ranges::default_sentinel_t{});
+        }
+
+        //!\copydoc operator!=
+        friend constexpr bool operator!=(std::ranges::default_sentinel_t const &,
+                                         async_buffer_iterator const & rhs) noexcept
+        {
+            return rhs != std::ranges::default_sentinel_t{};
+        }
+        //!\}
+    };
+
+public:
+    /*!\name Constructor, destructor, and assignment.
+     * \{
+     */
+    async_buffer_view()                                         = default; //!< Defaulted.
+    async_buffer_view(async_buffer_view const &)                = delete;  //!< Deleted.
+    async_buffer_view(async_buffer_view &&)                     = default; //!< Defaulted.
+    async_buffer_view & operator=(async_buffer_view const &)    = delete;  //!< Deleted.
+    async_buffer_view & operator=(async_buffer_view &&)         = default; //!< Defaulted.
+
+    /*!\brief Tell thread to stop and wait for it.
+     *
+     * \details
+     *
+     * Destruction of the view clears the buffer and temporary storage. Up to `n + 1` elements may be deleted
+     * together with the view. [This may be important if the underlying range is also single-pass and you
+     * are trying to estimate how many elements will be consumed – in case you do not plan on consuming the
+     * entire range.]
+     */
+    ~async_buffer_view()
+    {
+        stop_producer.store(true);
+        producer.join();
+    }
+
+    //!\brief Construction from the underlying view.
+    async_buffer_view(urng_t _urng, size_t const buffer_size) :
+        urange{std::move(_urng)}, buffer{buffer_size}, producer{&async_buffer_view::produce, this}
+    {}
+
+    //!\brief Construction from std::ranges::ViewableRange.
+    template <typename other_urng_t>
+    //!\cond
+    requires !std::Same<remove_cvref_t<other_urng_t>, async_buffer_view> && // prevent recursive instantiation
+             std::ranges::ViewableRange<other_urng_t> &&
+             std::Constructible<urng_t, ranges::ref_view<std::remove_reference_t<other_urng_t>>>
+    //!\endcond
+    async_buffer_view(other_urng_t && _urng, size_t const buffer_size) :
+        async_buffer_view{std::view::all(_urng), buffer_size}
+    {}
+    //!\}
+
+    /*!\name Iterators
+     * \{
+     */
+    /*!\brief Returns an iterator to the current begin of the underlying range.
+     *
+     * \details
+     *
+     * ### Thread-Safety
+     *
+     * It is thread-safe to call this function. Subsequent calls to begin will result in different
+     * iterators that are each valid individually. It is thread-safe to operate on different iterators
+     * from different threads (however it is not thread-safe to operate on a single iterator from different
+     * threads).
+     */
+    async_buffer_iterator begin()
+    {
+        return {buffer};
+    }
+
+    //!\brief Const version of begin is deleted, since the underlying view_state must be mutable.
```

Comment:
```suggestion
    //!\brief Const version of begin is deleted since the underlying view_state must be mutable.
```
(also for end).

Does begin link correctly?

Reply:
Most definitely not. In my defence: I copy'n'pasted this from @rrahn :wink: